tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -8361,7 +8361,45 @@ TEST_F(ExecutionTest, DynamicResourcesTest) {
     return;
   }
 
-  RunResourceTest(pDevice, pShader, L"cs_6_6", /*isDynamic*/true);
+  RunResourceTest(pDevice, pShader, L"cs_6_6", /*isDynamic*/false);
+  // Now test dynamic indexing
+
+  static const char pShaderDynamic[] =
+      "cbuffer Settings : register(b0, space0)"
+      "{"
+      "    uint InputIndex;"
+      "};"
+      "SamplerState PointClamp : register(s0, space0);"
+      "struct VSOutput {"
+      "  float2 Texture : TEXCOORD0;"
+      "};"
+      "float4 PSMain(VSOutput IN) : SV_TARGET"
+      "{"
+      "    Texture2D Input = ResourceDescriptorHeap[InputIndex];" // Dynamic
+      "    float3 color = Input.Sample(PointClamp, IN.Texture).rgb;"
+      "    color = color+1;"
+      "    return float4(color, 1.0f);"
+      "}\n";
```

Comment:
This is the right idea, but we need to match the expected outputs to reuse the verification code. For that, you'll want to model your shader after the `pShader` defined here https://github.com/microsoft/DirectXShaderCompiler/blob/ba2069d67a7045be8719e3a720c30e8a0e5bebcb/tools/clang/unittests/HLSL/ExecutionTest.cpp#L8329

The difference is that instead of the hard-coded literals to index into the `ResourceDescriptorHeap` and `SamplerDescriptorHeap`, you'll want to retrieve those same values from a cbuffer that you create by adding a function similar to `CreateSRV` https://github.com/microsoft/DirectXShaderCompiler/blob/ba2069d67a7045be8719e3a720c30e8a0e5bebcb/tools/clang/unittests/HLSL/ExecutionTest.cpp#L950 but for a constant buffer view (CBV). This page will be helpful: https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-createconstantbufferview

You'll want to populate that CBV with the same indices as are hard coded in `pShader` and use them just as they are in that shader.

You'll need to do the same with parameters to PMain. You could also dump them in `VSOutput` struct. You'll need to give them some annotation, you can use arbitrary annotations. I suggest you get the cbuffer variant done first though.